### PR TITLE
Add files via upload

### PR DESCRIPTION
**Pull request by DhillonTaran**
for issue #36

### CODE REVIEW

**Review by liamhyde02 — CHANGES_REQUESTED** [review]
Same as other prs, need to extend EntityParsingStep and use the EntityBuilder passed through the chain of responsibility

**Comment by khoa-l** [comment]
You're not merging into mainline? If you just want to move code, can you do that locally and just make changes on the issue branch instead? also can you make that branch name shorter? no need to have the entire issue in the branch name

File: src/main/java/org/codevillage/CompositionParsingStep.java [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+public class CompositionParsingStep extends EntityParsingStep {
+
+    public CompositionParsingStep(EntityParsingChain next) {
+        super(next);
+    }
+
+    @Override
+    public JavaEntity construct(EntityBuilder builder, CompilationUnit declaration) {
+        // Construct the entity using the builder
+        JavaEntity entity = builder.build();
```
Comment (buk0vec) [inline src/main/java/org/codevillage/CompositionParsingStep.java]: We don't call this until the last step

File: src/main/java/org/codevillage/CompositionParsingStep.java [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+        }
+
+        // Proceed with the next step in the chain, if any
+        if (next != null) {
+            return next.construct(builder, declaration);
+        }
+        return entity;
+    }
+
+    private List<String> getCompositionClassNames(JavaEntity entity) {
```
Comment (buk0vec) [inline src/main/java/org/codevillage/CompositionParsingStep.java]: This... doesn't work. You are getting the *literal class instance* of the JavaEntity and iterating through it. All these steps work around using the CompilationUnit that is passed in to determine things about the JavaEntity that we are trying to build. Please refer to how the other EntityParsingSteps work to see what I mean, the pattern will be similar to LOCParsingStep

File: src/main/java/org/codevillage/CompositionParsingStep.java [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+        List<String> compositions = getCompositionClassNames(entity);
+        for (String composition : compositions) {
+            builder.addComposition(composition);
+        }
+
+        // Proceed with the next step in the chain, if any
+        if (next != null) {
+            return next.construct(builder, declaration);
+        }
+        return entity;
```
Comment (buk0vec) [inline src/main/java/org/codevillage/CompositionParsingStep.java]: Not needed

**Review by buk0vec — CHANGES_REQUESTED** [review]
Please rewrite to actually pull compositions from the CompilationUnit!

File: src/main/java/org/codevillage/CompositionParsingStep.java (hunk 2 above)
Comment (buk0vec) [reply src/main/java/org/codevillage/CompositionParsingStep.java]: To clarify, JavaEntity is just a data class. We are calling methods on the builder as we find things out about the entity that we are building from the CompilationUnit (ex. name, realizations, etc). I'd recommend syncing-up with the rest of the repository and looking at the UML to see how everything works together.

**Review by buk0vec — APPROVED** [review]
lgtm, assuming you've tested these btw

**Review by buk0vec — APPROVED** [review]
lgtm idk i hope it works

**Review by liamhyde02 — APPROVED** [review]
lgtm

**Pull request merged**